src/lib/y2ntp_client/dialog/main.rb

```diff
@@ -0,0 +1,83 @@
+require "yast"
+
+require "cwm/dialog"
+require "y2ntp_client/widgets/main_widgets"
+
+Yast.import "Label"
+Yast.import "NtpClient"
+Yast.import "Stage"
+
+module Y2NtpClient
+  module Dialog
+    # Main entry point for Ntp Client
+    class Main < CWM::Dialog
+      def initialize
+        textdomain "ntp-client"
+      end
+
+      def title
+        # dialog caption
+        _("NTP Configuration")
+      end
+
+      def contents
+        replace_point = CWM::ReplacePoint.new(widget: CWM::Empty.new("empty_interval"))
+        table = Widgets::ServersTable.new
+        VBox(
+          HBox(
+            HWeight(1, Widgets::NtpStart.new(replace_point)),
+            HSpacing(1),
+            HWeight(1, replace_point)
+          ),
+          VSpacing(1),
+          Left(Widgets::PolicyCombo.new),
+          VSpacing(1),
+          *hardware_clock_widgets,
+          table,
+          HSpacing(0.2),
+          Left(
+            HBox(
+              Widgets::AddPoolButton.new,
+              HSpacing(),
+              Widgets::EditPoolButton.new(table),
+              HSpacing(),
+              Widgets::DeletePoolButton.new(table)
+            )
+          )
+        )
+      end
+
+      def run
+        loop do
+          res = super
+          return res if res != :redraw
+        end
+      end
+
+      def abort_button
+        Yast::Label.CancelButton
+      end
+
+      def hardware_clock_widgets
+        if Yast::NtpClient.ntp_conf.hardware_clock?
+          [
+            Label(_("Hardware clock configured as source. YaST will keep it untouched.")),
```

Comment:
(English is OK here, but) I don't understand what this is trying to tell me. So what? Will YaST not take a screwdriver and disassemble my hardware?

Reply:
Yes, I can guarantee it :). Maybe I should add `will keep this part of configuration untouched.`?